### PR TITLE
SSCS-10538 Post Hearing Request

### PR DESCRIPTION
**Pull request by TeaWithLucas**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10538

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/requestposthearing/RequestPostHearingSubmittedHandler.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class RequestPostHearingSubmittedHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+    private final CcdCallbackMapService ccdCallbackMapService;
+    @Value("${feature.postHearings.enabled}")
+    private final boolean isPostHearingsEnabled;
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/requestposthearing/RequestPostHearingSubmittedHandler.java:30]: do these need to be called in the handle method? (several of these in the new files)

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/requestposthearing/RequestPostHearingMidEventHandler.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        return response;
+    }
+
+    @Nullable
+    private static RequestFormat getRequestFormat(PostHearing postHearing) {
+        RequestPostHearingTypes typeSelected = postHearing.getRequestTypeSelected();
+        if (isNull(typeSelected)) {
+            return null;
+        }
+        switch (typeSelected) {
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/requestposthearing/RequestPostHearingMidEventHandler.java:66]: add space here

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/requestposthearing/RequestPostHearingAboutToSubmitHandler.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback,
+                                                          String userAuthorisation) {
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(caseData);
+
+        String caseId = caseData.getCcdCaseId();
+
+        RequestPostHearingTypes typeSelected = caseData.getPostHearing().getRequestTypeSelected();
+        log.info("Post Hearing Request: handing action {} for case {}", typeSelected,  caseId);
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/requestposthearing/RequestPostHearingAboutToSubmitHandler.java:45]: "handling"?

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/requestposthearing/RequestPostHearingSubmittedHandler.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                                                          String userAuthorisation) {
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(caseData);
+
+        Long caseId = Long.valueOf(caseData.getCcdCaseId());
+
+        PostHearing postHearing = caseData.getPostHearing();
+        RequestPostHearingTypes typeSelected = postHearing.getRequestTypeSelected();
+        log.info("Post Hearing Request: handing postHearing {} for case {}", typeSelected,  caseId);
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/requestposthearing/RequestPostHearingSubmittedHandler.java:50]: handling

**Review by amoslv1 — CHANGES_REQUESTED** [review]
Handful of lines without test coverage which you may (or may not) want to address.
https://sonarcloud.io/component_measures?id=SSCSTCA&metric=new_coverage&pullRequest=3060&view=list

Also some superficial typos in log messages.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/requestposthearing/RequestPostHearingSubmittedHandler.java (hunk 4 above)
Comment (TeaWithLucas) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/requestposthearing/RequestPostHearingSubmittedHandler.java]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/requestposthearing/RequestPostHearingAboutToSubmitHandler.java (hunk 3 above)
Comment (TeaWithLucas) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/requestposthearing/RequestPostHearingAboutToSubmitHandler.java]: done

**Review by nickhill111 — APPROVED** [review]

**Review by nickhill111 — APPROVED** [review]

**Review by amoslv1 — APPROVED** [review]

**Review by amoslv1 — APPROVED** [review]

**Review by nickhill111 — APPROVED** [review]

**Pull request merged**